doc/source/acb_theta.rst

```diff
@@ -0,0 +1,1540 @@
+
+    Such a worker will attempt to set *th* to the values `\theta_{a,0}(v,\tau)`
+    for `v = 0,t,2t,z,z+t,z+2t` and `a\in \{0,1\}^g` at shifted absolute
+    precision *prec*, and return `1` on success and `0` on failure. The vectors
+    *d0* and *d* must contain the result of :func:`acb_theta_dist_a0` on
+    `(0,\tau)` and `(z,\tau)`. If `z = 0`, `t = 0`, or both, we only compute
+    `3`, `2`, or `1` vectors of `2^g` values respectively.
+
+    Two functions of this type are available: :func:`acb_theta_ql_a0_naive` and
+    the main function :func:`acb_theta_ql_a0`. Using function pointers allows
+    us to write independent test code for the main workhorses
+    :func:`acb_theta_ql_a0_steps` and :func:`acb_theta_ql_a0_split` below.
+
+.. function:: int acb_theta_ql_a0_naive(acb_ptr th, acb_srcptr t, acb_srcptr z, arb_srcptr d0, arb_srcptr d, const acb_mat_t tau, slong guard, slong prec)
+
+    Follows the specifications of a function of type
+    :type:`acb_theta_ql_worker_t` using the naive algorithm only. The return
+    value is `1` iff the output vector *th* contains finite values.
+
+.. function:: int acb_theta_ql_a0_split(acb_ptr th, acb_srcptr t, acb_srcptr z, arb_srcptr d, const acb_mat_t tau, slong s, slong guard, slong prec, acb_theta_ql_worker_t worker)
+
+    Follows the specifications of a function of type
+    :type:`acb_theta_ql_worker_t`, except for the additional arguments *s* and
+    *worker*. We split the theta series according to the first `s` coordinates
+    of `n\in \mathbb{Z}^g`, writing `n = (n_0,n_1)` where `n_0\in \mathbb{Z}^s`
+    and `n_1\in \mathbb{Z}^{g - s}`. We must have `1\leq s\leq g -1`. Then
+    *worker* is called to evaluate the sum corresponding to each `n_1`. The
+    return value is 1 iff all the calls to *worker* succeed.
+
+    For each `0\leq a < 2^g`, we compute *R2* and *eps* as in
+    :func:`acb_theta_naive_radius` at shifted absolte precision *prec*. Note
+    that `n^T \mathrm{Im}(\tau) n\geq \lVert C_1 n_1\rVert^2`, where `C_1`
+    denotes the lower-right block of `C` of dimensions
+    `(g-s)\times(g-s)`. Thus, in order to compute `\theta_{a,0}(z, 2^n\tau)` at
+    shifted absolute precision *prec*, it is enough to consider those `n_1\in
+    \mathbb{Z}^{g - s}` in an ellipsoid `E_1` of radius *R2* for the Cholesky
+    matrix `C_1`. This ellipsoid is meant to contain very few points, and we
+    list all of them. Then, for a given choice of `n_1`, the sum of the
+    corresponding terms in the theta series is
+
+        .. math ::
+
+            e^{\pi i ((n_1 + \tfrac{a_1}{2})\tau_1 (n_1 + \tfrac{a_1}{2}) + 2 (n_1
+            + \tfrac{a_1}{2}) z_1)}
+            \theta_{a_0,0}(z_0 + x (n_1 + \tfrac{a_1}{2}), \tau_0).
+
+    where `\tau = (\begin{smallmatrix} \tau_0 & x\\x^T &
+    \tau_1\end{smallmatrix})` and `z = (z_0,z_1)`. When calling *worker*, we
```

Comment:
If `psmallmatrix` does not exist, perhaps use `\bigl(` and `\bigr)` as delimiters?